python/ray/experimental/channel.py

```diff
@@ -185,9 +301,13 @@ def close(self) -> None:
         Does not block. Any existing values in the channel may be lost after the
         channel is closed.
         """
-        logger.debug(f"Setting error bit on channel: {self._base_ref}")
-        self.ensure_registered_as_writer()
-        self._worker.core_worker.experimental_channel_set_error(self._base_ref)
+        logger.debug(f"Setting error bit on channel: {self._writer_ref}")
+        try:
+            self.ensure_registered_as_reader()
+            # TODO: Also close on the reader ref?
```

Comment:
Hmm actually I'm kind of surprised this works without closing the reader ref... can we add a unit test that checks DAG teardown for the multinode case where the reader and writer refs are different?

Reply:
Done. I also added `test_remote_reader_close` to test_channel.py.